DANS-KNAW/dd-pre-publish-workflow
# DD-131 - Create pre-publication workflow to mint PIDs

See: https://drivenbydata.atlassian.net/browse/DD-131

To test do the following:  

1. copy workflow from `src/main/scala/nl.knaw.dans.dd.wf/workflows/pid-workflow.json` to dd-dtap `docs` folder

2. In a running dd-dtap instance:  

* run `vagrant ssh`
* Register workflow `curl http://localhost:8080/api/admin/workflows -X POST --data-binary @/vagrant/docs/pid-workflow.json -H "Content-type: application/json"`
* Set as default pre-publish workflow `curl http://localhost:8080/api/admin/workflows/default/PrePublishDataset -X PUT -d 2`
* Add host ip-address to whitelist for resume requests `curl -X PUT -d "127.0.0.1;192.168.33.1" http://localhost:8080/api/admin/workflows/ip-whitelist`
note: I'm not sure if the hosts ip address is always 192.168.33.1

3. In dd-easy-workflows-poc (to be renamed):
* Build project
* Add api token to `home/cfg/application.properties`
* run `run-service.sh` ( or `run-service-debug.sh`)

Add Dataset in Dataverse. Upon publication the workflow should be triggered.
